src/System.Text.Json/src/System/Text/Json/Node/JsonNode.cs

```diff
@@ -10,5 +16,105 @@ namespace System.Text.Json
+        {
+            if (!jsonElement.IsImmutable)
+            {
+                jsonNode = (JsonNode)jsonElement._parent;
+                return true;
+            }
+
+            jsonNode = null;
+            return false;
+        }
+
+        public static JsonNode Parse(string json)
+        {
+            using (JsonDocument document = JsonDocument.Parse(json))
+            {
+                return DeepCopy(document);
+            }
+        }
+
+        public static JsonNode Parse(ReadOnlySequence<byte> utf8Json)
+        {
+            using (JsonDocument document = JsonDocument.Parse(utf8Json))
+            {
+                return DeepCopy(document);
+            }
+        }
+
+        public static JsonNode Parse(Stream utf8Json)
+        {
+            using (JsonDocument document = JsonDocument.Parse(utf8Json))
+            {
+                return DeepCopy(document);
+            }
+        }
+
+        public static JsonNode Parse(ReadOnlyMemory<byte> utf8Json)
+        {
+            using (JsonDocument document = JsonDocument.Parse(utf8Json))
+            {
+                return DeepCopy(document);
+            }
+        }
+
+        public static JsonNode Parse(ReadOnlyMemory<char> json)
+        {
+            using (JsonDocument document = JsonDocument.Parse(json))
+            {
+                return DeepCopy(document);
+            }
+        }
+
+        public static JsonNode DeepCopy(JsonNode jsonNode) => jsonNode.Clone();
+
+        public abstract JsonNode Clone();
+
+        public static JsonNode DeepCopy(JsonElement jsonElement)
+        {
+            switch (jsonElement.ValueKind)
+            {
+                case JsonValueKind.Object:
+                    JsonObject jsonObject = new JsonObject();
+                    foreach (JsonProperty property in jsonElement.EnumerateObject())
+                    {
+                        jsonObject.Add(property.Name, DeepCopy(property.Value));
+                    }
+                    return jsonObject;
+                case JsonValueKind.Array:
+                    JsonArray jsonArray = new JsonArray();
+                    foreach (JsonElement element in jsonElement.EnumerateArray())
+                    {
+                        jsonArray.Add(DeepCopy(element));
+                    }
+                    return jsonArray;
+                case JsonValueKind.Number:
+                    return new JsonNumber(jsonElement.GetRawText());
+                case JsonValueKind.String:
+                    return new JsonString(jsonElement.GetString());
+                case JsonValueKind.True:
+                    return new JsonBoolean(true);
+                case JsonValueKind.False:
+                    return new JsonBoolean(false);
+                case JsonValueKind.Null:
+                    return null;
+                default:
+                    throw new ArgumentException();
```

Comment:
ArgumentExceptions should have a message and identify the parameter by name via the `paramName` constructor argument.  (There's probably one somewhere about `default(JsonElement)`, which is the only case that can hit this.)